nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -1670,6 +1670,49 @@ public ProcessorDTO getConfiguration() {
     }
 
     @Override
+    public ConfigurationSnapshot<ProcessorDTO> setProcessorProperties(final Revision revision, final String processorId, final Map<String,String> properties) {
```

Comment:
I think it may make sense to just use updateProcessor(...) instead introducing setProcessorProperties(...). We should probably even get rid of setProcessorAnnotationData(...) in favor of updateProcessor(...). In earlier version of NiFi, these values were setting differently which is why there was two different method. However, now I think we can consolidate these.